kernprof.py

```diff
@@ -248,9 +246,23 @@ def _restore_list(lst):
     >>> l
     [1, 2, 3]
     """
-    old = lst.copy()
-    yield
-    lst[:] = old
+    def __init__(self, lst):
+        self.lst = lst
+        self.contexts = []
+
+    def __enter__(self):
+        self.contexts.append(self.lst.copy())
+
+    def __exit__(self, *_, **__):
+        self.lst[:] = self.contexts.pop()
```

Comment:
```suggestion
        self.lst = lst
        self.old = None

    def __enter__(self):
        assert self.old is None
        self.old = self.lst.copy()

    def __exit__(self, *_, **__):
        self.lst[:] = self.old
```

Let's just make this map onto the original. No need to handle nested versions of the same context manager. If you want, we can raise an exception to make sure reentry / reuse doesn't happen.

Reply:
Makes sense, our usage of the context manager is limited enough that reentry shouldn't be an issue.